charms/slurmctld/src/charm.py

```diff
@@ -90,36 +84,30 @@ def __init__(self, *args):
 
     def _on_install(self, event: InstallEvent) -> None:
         """Perform installation operations for slurmctld."""
-        self.unit.status = WaitingStatus("Installing slurmctld")
-
-        if self._slurmctld_manager.install():
-
-            # Store the munge_key and jwt_rsa key in the stored state.
-            # NOTE: Use secrets instead of stored state when secrets are supported the framework.
-            if self.model.unit.is_leader():
-                jwt_rsa = self._slurmctld_manager.generate_jwt_rsa()
-                self._stored.jwt_rsa = jwt_rsa
-
-                munge_key = self._slurmctld_manager.generate_munge_key()
-                self._stored.munge_key = munge_key
-
-                self._slurmctld_manager.stop_munged()
-                self._slurmctld_manager.write_munge_key(munge_key)
-                self._slurmctld_manager.start_munged()
-
-                self._slurmctld_manager.stop_slurmctld()
-                self._slurmctld_manager.write_jwt_rsa(jwt_rsa)
-                self._slurmctld_manager.start_slurmctld()
+        self.unit.status = WaitingStatus("installing slurmctld")
+        try:
+            if self.unit.is_leader():
+                self._slurmctld.install()
+
+                # TODO: <make github issue> -
```

Comment:
Is there an issue for this? I think it's still a valid issue even without this PR merged.

Reply:
Oops, this was a `TODO` comment I made early on during the work on this PR. I'll go create an issue and update the comment.

Reply:
Cleaned up all the TODO comments in https://github.com/charmed-hpc/slurm-charms/pull/35/commits/2ba972fdec822e6a3b382951670d279716435f3d